### PR TITLE
feat: add list user invitations api and add orgID filter in list user groups api

### PR DESCRIPTION
**Pull request by Chief-Rishab**
- Require https://github.com/raystack/proton/pull/281

### CODE REVIEW

**Comment by kushsharma** [comment]
Can we split these docs changes in different PR? What exactly is this?

**Comment by Chief-Rishab** [comment]
@kushsharma comes from the recent name change from odpf to raystack in proton files

File: internal/store/postgres/invitation.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func (from Invitation) transformToInvitation() (invitation.Invitation, error) {
 		}
 	}
 	var groupIDs []string
-	if val, ok := unmarshalledMetadata["group_ids"]; ok {
+	if val, ok := unmarshalledMetadata["group_ids"]; ok && (val != nil) {
```
Comment (kushsharma) [inline internal/store/postgres/invitation.go:29]: Do you think this check was unnecessary? I don't remember the case, but I saw `val` being null in sometimes.

File: internal/store/postgres/invitation_repository.go [hunk 2]
```diff
@@ -115,11 +115,43 @@ func (s *InvitationRepository) List(ctx context.Context, flt invitation.Filter)
 
 	query, params, err := stmt.ToSQL()
 	if err != nil {
-		return nil, fmt.Errorf("%w: %s", queryErr, err)
+		return []invitation.Invitation{}, fmt.Errorf("%w: %s", queryErr, err)
```
Comment (kushsharma) [inline internal/store/postgres/invitation_repository.go:118]: Why not `nil`?

File: internal/store/postgres/invitation.go (hunk 1 above)
Comment (Chief-Rishab) [reply internal/store/postgres/invitation.go:29]: I created a user invitatioon just for an org, (without any group_ids) and listed the pending invitations. There was an error for trying to unmarshall the null value in the server logs

Comment (Chief-Rishab) [reply internal/store/postgres/invitation.go:29]: Exact error `interface conversion: interface {} is nil, not []interface {}`

File: internal/store/postgres/invitation_repository.go (hunk 2 above)
Comment (Chief-Rishab) [reply internal/store/postgres/invitation_repository.go:118]: yes, can revert it to nil

File: internal/store/postgres/invitation.go (hunk 1 above)
Comment (kushsharma) [reply internal/store/postgres/invitation.go:29]: Cool, make sense

**Review by kushsharma — APPROVED** [review]
LGTM

**Pull request merged**